File: ErshovVladislav/Task4/Task4/src/main/java/ershov/examsystem/ExamSystem.java
package ershov.examsystem;

import java.util.Map;
import syncronizedset.ISynchronizedSet;
import syncronizedset.OptimisticSet;

public class ExamSystem implements IExamSystem {

	private final ISynchronizedSet<Map.Entry<Long, Long>> examResults = new OptimisticSet<>();

	@Override
	public void add(long studentId, long courseId) {
		examResults.add(Map.entry(studentId, courseId));
	}

	@Override
	public void remove(long studentId, long courseId) {
		examResults.remove(Map.entry(studentId, courseId));
	}

	@Override
	public boolean contains(long studentId, long courseId) {
		return examResults.contains(Map.entry(studentId, courseId));
	}

	@Override
	public int count() {
		return examResults.size();
	}

}

File: ParshinMaksim/Task4/Task4/ExamSystem.cs
namespace Task4
{
	public class ExamSystem : IExamSystem
	{
		private readonly ISet<ExamInfo> set;

		public ExamSystem(Func<ISet<ExamInfo>> setFactory)
		{
			set = setFactory();
		}

		public int Count => set.Count;

		public void Add(long studentId, long courseId) => set.Add(new ExamInfo(studentId, courseId));

		public bool Contains(long studentId, long courseId) => set.Contains(new ExamInfo(studentId, courseId));

		public void Remove(long studentId, long courseId) => set.Remove(new ExamInfo(studentId, courseId));
	}
}
